testresultparser/parser.go

```diff
@@ -0,0 +1,180 @@
+package testresultparser
+
+import (
+	"encoding/xml"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// MVNFailure captures failure noted in Test Case in XML Report
+type MVNFailure struct {
+	FailureType string `xml:"type,attr"`
+	Message     string `xml:",chardata"`
+}
+
+// TestCase records test case info in XML Report
+type TestCase struct {
+	Name string     `xml:"name,attr"`
+	F    MVNFailure `xml:"failure"` // struct for Failure Details
+}
+
+// TestResultXML is struct for XML Test Report
+type TestResultXML struct {
+	TestSuite string     `xml:"name,attr"`
+	Tests     int        `xml:"tests,attr"`
+	Failures  int        `xml:"failures,attr"`
+	Errors    int        `xml:"errors,attr"`
+	Skipped   int        `xml:"skipped,attr"`
+	Time      string     `xml:"time,attr"`
+	T         []TestCase `xml:"testcase"` // struct for each Test Case
+}
+
+// failsafeReportXML is struct for Failafe Reports
+type failsafeReportXML struct {
+	Result         string `xml:"result,attr"`
+	TimeOut        string `xml:"timeout,attr"`
+	Tests          int    `xml:"completed"`
+	Failures       int    `xml:"failures"`
+	Errors         int    `xml:"errors"`
+	Skipped        int    `xml:"skipped"`
+	FailureMessage string `xml:"failureMessage"`
+}
+
+// Failure struct is the part of the Result struct for the output of the Parser
+type Failure struct {
+	TestCase string
+	Type     string
```

Comment:
I don't understand how Failure struct could be enum/iota ? The values aren't constants, neither are they auto-incremental. Please correct me, if I am wrong here.